android/modules/ui/src/java/ti/modules/titanium/ui/TabGroupProxy.java

```diff
@@ -384,50 +386,82 @@ public void setTitle(String title)
 	@Override
 	protected void handleOpen(KrollDict options)
 	{
-		Activity topActivity = TiApplication.getAppCurrentActivity();
 		// Don't open if app is closing or closed
-		if (topActivity == null || topActivity.isFinishing()) {
+		Activity topActivity = getActivity();
+		if (topActivity == null || topActivity.isFinishing() || topActivity.isDestroyed()) {
 			return;
 		}
 
-		// set theme for XML layout
-		if (hasProperty(TiC.PROPERTY_STYLE)
-			&& ((Integer) getProperty(TiC.PROPERTY_STYLE)) == AndroidModule.TABS_STYLE_BOTTOM_NAVIGATION
-			&& getProperty(TiC.PROPERTY_THEME) != null) {
-			try {
-				String themeName = getProperty(TiC.PROPERTY_THEME).toString();
-				int theme = TiRHelper.getResource("style."
-					+ themeName.replaceAll("[^A-Za-z0-9_]", "_"));
-				topActivity.setTheme(theme);
-				topActivity.getApplicationContext().setTheme(theme);
-			} catch (Exception e) {
+		/**
+		 * Only applicable on Android 12 or Android 13.
+		 * As reported here on Google: https://issuetracker.google.com/issues/293645024
+		 * WindowContainer may not be available in very rare cases on devices running Android 12/13.
+		 * Wait for the view to be attached to a window before opening the TabGroup.
+		 */
+		if (Build.VERSION.SDK_INT >= 31 && Build.VERSION.SDK_INT <= 33) {
+			Window window = topActivity.getWindow();
+			View decorView = window != null ? window.getDecorView() : null;
+			if (decorView == null) {
+				return;
+			}
+
+			if (decorView.getDisplay() == null) {
+				TiSafeDisplay.getDisplaySafely(decorView, isDisplayAvailable -> {
+					// Display may not be available in very very rare cases,
+					// but we open it since it's now in try-catch.
+					openTabGroup(topActivity, options);
```

Comment:
What's the typical delay in ms for this case? And should we include it in 12.7.0 tomorrow?

Reply:
Didn't check for the delay as it was not happening on my devices. Checking for delay won't be a proper solution as Android 12/13 devices not having this issue will also go under the delay.
Yes, let's include it in 12.7.0.

Reply:
Ok, got it. I think @m1ga help us know it to check the time difference between a new log in `index.js` and the log `checkpoint, app created` posted by SDK already. Android Studio shows timestamp of the logs in its `Logcat`.